src/com/Mustang.cpp

```diff
@@ -27,19 +28,39 @@
 
 namespace plug::com
 {
-    SignalChain decode_data(const std::array<PacketRawType, 7>& data)
+    SignalChain decode_data(const std::array<PacketRawType, 7>& data, DeviceModel model)
     {
-        const auto name = decodeNameFromData(fromRawData<NamePayload>(data[0]));
-        const auto amp = decodeAmpFromData(fromRawData<AmpPayload>(data[1]), fromRawData<AmpPayload>(data[6]));
-        const auto effects = decodeEffectsFromData({{fromRawData<EffectPayload>(data[2]), fromRawData<EffectPayload>(data[3]),
-                                                     fromRawData<EffectPayload>(data[4]), fromRawData<EffectPayload>(data[5])}});
-
-        return SignalChain{name, amp, effects};
+        switch(model.category())
+        {
+            case DeviceModel::Category::MustangV1:
+            case DeviceModel::Category::MustangV2:
+                {
+                    const auto name = decodeNameFromData(fromRawData<NamePayload>(data[0]));
+                    const auto amp = decodeAmpFromData(fromRawData<AmpPayload>(data[1]), fromRawData<AmpPayload>(data[6]));
+                    const auto effects = decodeEffectsFromData({{fromRawData<EffectPayload>(data[2]), fromRawData<EffectPayload>(data[3]),
+                                                                fromRawData<EffectPayload>(data[4]), fromRawData<EffectPayload>(data[5])}});
+
+                    return SignalChain{name, amp, effects};
+                }
+
+            case DeviceModel::Category::MustangV3_USB:
+                {
+                    const auto name = decodeNameFromData(fromRawData<NamePayload>(data[0]));
```

Comment:
It is now clear that the package reflected in data[0] does not contain anything useful for building a name.

Hopefully the name is available somewhere in the JSON we expect to be able to extract in the next iteration of this change.